pkg/operator/hive/hive_controller.go

```diff
@@ -384,13 +382,21 @@ func (r *ReconcileHiveConfig) Reconcile(ctx context.Context, request reconcile.R
 		Namespace: r.hiveOperatorNamespace,
 	})
 
+	h, err := resource.NewHelperFromRESTConfig(r.restConfig, hLog)
+	if err != nil {
+		hLog.WithError(err).Error("error creating resource helper")
+		instance.Status.Conditions = util.SetHiveConfigCondition(instance.Status.Conditions, hivev1.HiveReadyCondition, corev1.ConditionFalse, "ErrorCreatingResourceHelper", err.Error())
+		r.updateHiveConfigStatus(origHiveConfig, instance, hLog, false)
```

Comment:
Agree. This was just moved up verbatim from [below](https://github.com/openshift/hive/pull/1601/files#diff-f2392ee1d4eaa422834a890dbff18dd21413c4e31d617324b8b350ef83b9bf26L436) -- but will fix.

Reply:
[Later] Looking more closely at this, the pattern exists throughout the controller. `updateHiveConfigStatus` does in fact [log](https://github.com/openshift/hive/blob/master/pkg/operator/hive/hive_controller.go#L663) if the status update fails. And since we're already in an error branch in all these cases, we're still returning an error from Reconcile, which will trigger the right behavior (a requeue). I can only guess the original authors considered it more useful for the reconcile to report the original error rather than the status update error.

If we still think this logic should be changed somehow, can we discuss it for a separate PR?

Reply:
Yep, I see that now. Perhaps the right thing is collecting the failure into an aggregated error, but that seems out of scope for this PR.